zmon_aws_agent/postgresql.py

```diff
@@ -58,3 +59,124 @@ def get_databases_from_clusters(pgclusters, infrastructure_account, region,
         logger.exception("Failed to make Database entities for PostgreSQL clusters!")
 
     return entities
+
+
+def collect_addresses(infrastructure_account):
+    ec2 = boto3.client('ec2')
+
+    addresses = call_and_retry(ec2.describe_addresses)['Addresses']
+
+    return [a for a in addresses if a['NetworkInterfaceOwnerId'] == infrastructure_account.split(':')[1]]
+# FIXME: depend on region, too?
+
+
+def collect_asgs(infrastructure_account):
+    asg = boto3.client('autoscaling')
+
+    asg_paginator = asg.get_paginator('describe_auto_scaling_groups')
+    all_groups = call_and_retry(lambda: asg_paginator.paginate().build_full_result()['AutoScalingGroups'])
+
+    return [gr for gr in all_groups
+            if gr['AutoScalingGroupARN'].split(':')[4] == infrastructure_account.split(':')[1]
+            and ('Key', 'SpiloCluster') in [i for t in [g.items() for g in gr['Tags']] for i in t]]
+
+
+def collect_instances(infrastructure_account):
+    ec2 = boto3.client('ec2')
+
+    inst_paginator = ec2.get_paginator('describe_instances')
+    instances = inst_paginator.paginate().build_full_result()['Reservations']
+
+    # we assume only one instance per reservation
+    return [i['Instances'][0] for i in instances if i['OwnerId'] == infrastructure_account.split(':')[1]]
+
+
+def collect_launch_configurations(infrastructure_account):
```

Comment:
... another call to 'autoscaling'

Reply:
but this is `describe_launch_configurations`, in any case called in a rare edge case only.